### PR TITLE
zsh-fzf-tab: init at 2021-01-24

### PR DESCRIPTION
**Pull request by Vonfry**
###### Motivation for this change
A zsh plugins.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/shells/zsh/zsh-fzf-tab/default.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  installPhase = ''
+     mkdir -p ${INSTALL_PATH}
+     cp -r lib ${INSTALL_PATH}/lib
+     install -D fzf-tab.zsh ${INSTALL_PATH}/fzf-tab.zsh
+     install -D fzf-tab.plugin.zsh ${INSTALL_PATH}/fzf-tab.plugin.zsh
+     install -D modules/Src/aloxaf/fzftab.so ${INSTALL_PATH}/modules/Src/aloxaf/fzftab.so
+  '';
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/shells/zsh/zsh-fzf-tab/default.nix:38]:
```suggestion
  meta = with lib; {
```

File: pkgs/shells/zsh/zsh-fzf-tab/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    homepage = "https://github.com/Aloxaf/fzf-tab";
+    description = "Replace zsh's default completion selection menu with fzf!";
+    license = licenses.mit;
+    maintainers = with maintainers; [ vonfry ];
+    platforms = platforms.linux;
```
Comment (SuperSandro2000) [inline pkgs/shells/zsh/zsh-fzf-tab/default.nix:43]: Should this work on darwin?

File: pkgs/shells/zsh/zsh-fzf-tab/default.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  postConfigure = ''
+    pushd modules
+    ./configure --disable-gdbm --without-tcsetpgrp
+    popd
+  '';
+
+  postBuild = ''
+    pushd modules
+    make -j
```
Comment (SuperSandro2000) [inline pkgs/shells/zsh/zsh-fzf-tab/default.nix:26]:
```suggestion
    make -j$NIX_BUILD_CORES
```

File: pkgs/shells/zsh/zsh-fzf-tab/default.nix [hunk 4]
```diff
@@ -0,0 +1,45 @@
+{ stdenv, fetchFromGitHub, ncurses }:
+
+let
+     INSTALL_PATH="${placeholder "out"}/share/fzf-tab";
+in stdenv.mkDerivation rec {
+  pname = "zsh-fzf-tab";
+  version = "2021-01-24";
```
Comment (SuperSandro2000) [inline pkgs/shells/zsh/zsh-fzf-tab/default.nix:7]:
```suggestion
  version = "unstable-2021-01-24";
```

File: pkgs/shells/zsh/zsh-fzf-tab/default.nix (hunk 2 above)
Comment (Vonfry) [reply pkgs/shells/zsh/zsh-fzf-tab/default.nix:43]: It needs a more argument when configuring, but I cannot test it.

File: pkgs/shells/zsh/zsh-fzf-tab/default.nix [hunk 5]
```diff
@@ -0,0 +1,45 @@
+{ stdenv, lib, fetchFromGitHub, ncurses }:
+
+let
+     INSTALL_PATH="${placeholder "out"}/share/fzf-tab";
```
Comment (SuperSandro2000) [inline pkgs/shells/zsh/zsh-fzf-tab/default.nix:4]:
```suggestion
  INSTALL_PATH="${placeholder "out"}/share/fzf-tab";
```

**Pull request merged**